### PR TITLE
Support ":" in node name for mean/scale application

### PR DESCRIPTION
**Pull request by mvafin**
Description: Support ":" in node names when using mean/scale in MO

JIRA: 42977

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py - N/A no new models added
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Supported **public** models list - N/A
* [x]  New operations specification - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: model-optimizer/extensions/middle/AddMeanScaleValues.py [hunk 1]
```diff
@@ -91,28 +91,27 @@ def find_and_replace_pattern(self, graph: Graph):
...
 
-        for node_name in values:
-            node_mean_scale_values = values[node_name]
-            node_name, port = split_node_in_port(node_name)
+        for node_name, node_mean_scale_values in values.items():
             node_id = None
             try:
-                node_id = graph.get_node_id_by_name(node_name)
+                node_id, direction, port = get_node_id_with_ports(graph, node_name)
+                assert direction != 'out', 'Only input port can be specified for mean/scale application'
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/AddMeanScaleValues.py:114]: nit: may be it is a bit easier to understand if the assert will be like this `assert direction == 'in'`

File: model-optimizer/extensions/middle/AddMeanScaleValues_test.py [hunk 2]
```diff
@@ -255,6 +255,24 @@ def test_mean_values_explicit_and_scale_values_explicit_with_shape_of(self):
         self.assertTrue(flag, resp)
         self.check_graph_attrs(graph, graph_ref, ['parameter'])
 
+    def test_mean_values_with_double_dots_in_node_name(self):
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/AddMeanScaleValues_test.py:258]: `double_dots` -> `colon`

File: model-optimizer/extensions/middle/AddMeanScaleValues_test.py [hunk 3]
```diff
@@ -255,6 +255,24 @@ def test_mean_values_explicit_and_scale_values_explicit_with_shape_of(self):
...
+        graph = build_graph(nodes, [*connect('parameter', 'result')], {'parameter': {'name': 'param:0'}},
+                            nodes_with_edges_only=True, cli=argv)
+        self.set_graph_attrs(graph, ['parameter'])
+        self.set_graph_attrs(graph_ref, ['parameter'])
+        graph.graph['layout'] = 'NCHW'
+
+        AddMeanScaleValues().find_and_replace_pattern(graph)
+        (flag, resp) = compare_graphs(graph, graph_ref, 'result', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/AddMeanScaleValues_test.py:275]: Please, add a test where the mean/scale values are specified using input port.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/middle/AddMeanScaleValues.py (hunk 1 above)
Comment (mvafin) [reply model-optimizer/extensions/middle/AddMeanScaleValues.py:114]: Direction can be None if no port provided and it is valid for this code.

**Review by vgavrilo — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**